uppsrc/CtrlCore/X11Proc.cpp

```diff
@@ -418,17 +419,20 @@ void Ctrl::EventProc(XWindow& w, XEvent *event)
 		}
 		break;
 	case MotionNotify:
-		while(XCheckWindowEvent(Xdisplay, top->window, PointerMotionMask, event));
-		EndIgnore();
-		mousePos = Point(event->xmotion.x_root, event->xmotion.y_root);
-		Xeventtime = event->xmotion.time;
-		Point p = mousePos - Xbuttonpos;
-		if(max(abs(p.x), abs(p.y)) > 4)
-			Xbuttontime = Xeventtime - 0x80000000;
-		sModState = event->xmotion.state;
-		DispatchMouse(MOUSEMOVE, Point(event->xmotion.x, event->xmotion.y));
-		DoCursorShape();
-		break;
+		Top *top = GetTop();
+		if(top) {
+			while(XCheckWindowEvent(Xdisplay, top->window, PointerMotionMask, event));
+			EndIgnore();
+			mousePos = Point(event->xmotion.x_root, event->xmotion.y_root);
+			Xeventtime = event->xmotion.time;
+			Point p = mousePos - Xbuttonpos;
+			if(max(abs(p.x), abs(p.y)) > 4)
+				Xbuttontime = Xeventtime - 0x80000000;
+			sModState = event->xmotion.state;
+			DispatchMouse(MOUSEMOVE, Point(event->xmotion.x, event->xmotion.y));
+			DoCursorShape();
+			break;
```

Comment:
break should be after if statement. In this case it doesn't matter, but when there will be additiona case after MotionNotify it might lead to unpredictable behavior.

However, as I suggested we should use early return pattern here instead:
```
if (top) {
    break;
}

while(XCheckWindowEvent(Xdisplay, top->window, PointerMotionMask, event));
....
break;
```

Reply:
Yes.